libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -3763,8 +3772,14 @@ reify ::
   (a -> b) ->
   (Term fn b -> p) ->
   Pred fn
-reify tm f body =
-  mkReify tm f (bind body)
+reify t f body =
+  exists (\eval -> pure $ f (eval t)) $ \x ->
+    [ reifies x t f
+    , toPred $ body x
+    ]
+
+reifies :: (HasSpec fn a, HasSpec fn b) => Term fn b -> Term fn a -> (a -> b) -> Pred fn
+reifies = Reifies
```

Comment:
I guess this is the heart of the matter. 'reify' has the same type that it had before. Except it is implemented in terms of 
exists and the new Pred 'Reifies'

Reply:
Yes, and the point of that is that it reduces the number of different binding-forms that we have to consider. This simplifies dependency resolution and allows examples like `reifyYucky` to pass.